### PR TITLE
Add some leftovers from the update to 2.6

### PR DESCRIPTION
**Pull request by Morriar**
### Motivation

While running the [import_whitequark](https://github.com/sorbet/sorbet/blob/master/tools/scripts/import_whitequark.sh) script for 2.7 I grabbed a few tests belonging to 2.6 so let's add them before doing the bump.

### Test plan

See included automated tests.

### CODE REVIEW

File: third_party/parser/include/ruby_parser/lexer.hh [hunk 1]
```diff
@@ -26,6 +26,7 @@ enum class ruby_version {
     RUBY_23,
     RUBY_24,
     RUBY_25,
+    RUBY_26,
```
Comment (Morriar) [inline third_party/parser/include/ruby_parser/lexer.hh:29]: This is useless as there are no conditionals based on 2.6 version in the lexer right now but I will add `RUBY_27` after and it seemed odd to jump a version.

File: third_party/parser/cc/driver.cc [hunk 2]
```diff
@@ -16,7 +16,7 @@ base_driver::base_driver(ruby_version version, const std::string& source, const
 }
 
 typedruby25::typedruby25(const std::string& source, const struct builder& builder)
-	: base_driver(ruby_version::RUBY_25, source, builder)
+	: base_driver(ruby_version::RUBY_26, source, builder)
```
Comment (Morriar) [inline third_party/parser/cc/driver.cc:19]: Is not changing anything.

File: third_party/parser/cc/driver.cc [hunk 3]
```diff
@@ -16,7 +16,7 @@ base_driver::base_driver(ruby_version version, const std::string& source, const
 }
 
 typedruby25::typedruby25(const std::string& source, const struct builder& builder)
```
Comment (Morriar) [inline third_party/parser/cc/driver.cc:18]: Should we also rename this class? My understanding is that we're aiming to a kind of version-agnostic parser with Sorbet.

Comment (jez) [reply third_party/parser/cc/driver.cc:18]: feel free! it's probably just an oversight.

**Pull request merged**